console/console-init/ui/src/modules/iot-device/components/CreateMetadata/MetaDataRow.tsx

```diff
@@ -29,19 +32,47 @@ export const MetaDataRow: React.FC<IMetaDataRow> = ({
   setMetadataList,
   rowIndex
 }) => {
-  const metadataRow = metadataList[rowIndex];
+  const currentRow = metadataList[rowIndex];
   const [validationStatus, setValidationStatus] = useState<
     ValidationStatusType.DEFAULT | ValidationStatusType.ERROR
   >(ValidationStatusType.DEFAULT);
 
-  const onSelectType = (typeValue: string) => {
-    const validationStatus = getValidationStatus(typeValue, metadataRow.value);
+  const updateMetadataList = (property: string, value: string) => {
+    let updatedMetadataList = [...metadataList];
+    switch (property) {
```

Comment:
I think we don't need switch case here. You can try this single line of code.
updatedMetadataList[rowIndex][property]=isObjectOrArray(value as any) ? "" : value;

Reply:
You're right, will change.

Reply:
Thanks for the suggestion, but this is not entirely the case here. The isObjectOrArray is used only in case of types and also the empty value is not being assigned to the `property`. I am refactoring this block to use `updatedMetadataList[rowIndex][property]` and instead handling the case with `if`. Let me know if this makes sense.